mne/viz/ica.py

```diff
@@ -206,19 +240,49 @@ def _plot_ica_sources_evoked(evoked, exclude, title, show):
     times = evoked.times * 1e3
 
     # plot unclassified sources
-    plt.plot(times, evoked.data.T, 'k')
-    for ii in exclude:
-        # use indexing to expose event related sources
-        plt.plot(times, evoked.data[ii].T, color='r', label='ICA %02d' % ii)
+    lines = list()
+    texts = list()
+    labels = list()
+    if picks is None:
+        picks = np.arange(evoked.data.shape[0])
+    # plt.plot(times, evoked.data[picks].T, 'k')
+    for ii in picks:
+        if ii in exclude:
+            label = 'ICA %02d' % ii
```

Comment:
ok, but in the original code (https://github.com/mne-tools/mne-python/pull/2146/files#diff-0b10f8094c279e08b33dd3d31e23cc76L212), that's the labeling that was used. we should make it clear to the user that index number is not the same the ICA name.